gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/ClientCredentialsResource.java

```diff
@@ -40,6 +54,47 @@ public class ClientCredentialsResource extends TokenResource {
     public static final String CLIENT_ID = "client_id";
     public static final String CLIENT_SECRET = "client_secret";
 
+    private GatewayServices services;
+
+    @Override
+    protected ServletContext wrapContextForDefaultParams(ServletContext context) throws ServletException {
+        ServletContext wrapperContext = new ServletContextWrapper(context);
+        wrapperContext.setInitParameter(TokenStateService.CONFIG_SERVER_MANAGED, "true");
+        wrapperContext.setInitParameter(TokenResource.TOKEN_TTL_PARAM, "-1");
```

Comment:
This could never be set otherwise then, right?

Reply:
Topology level config overrides the defaults.